### PR TITLE
Add ruff.toml and clean up warnings

### PR DESCRIPTION
**Pull request by nathanchance**
[`ruff`](https://github.com/charliermarsh/ruff) is a fast Python linter written in Rust. Add a `ruff.toml` file to turn on several classes of warnings that are useful and fix them up.

### CODE REVIEW

File: ruff.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+select = [
```
Comment (nickdesaulniers) [inline ruff.toml:1]: Mind adding a comment linking to https://beta.ruff.rs/docs/rules/? That should make it faster to find the codes in the future

**Review by nickdesaulniers — COMMENTED** [review]
The commit message `Fix 'ruff check --fix' for COM812` doesn't really explain what's going on. Got the warning text handy?

File: generate.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def parse_args(trees):
                         help='Fail if generating yml files results in a diff')
     parser.add_argument(
         'trees',
-        choices=trees + ['all'],
+        choices=[*trees, 'all'],
```
Comment (nickdesaulniers) [inline generate.py:25]: What is this? Neat.

File: check_logs.py [hunk 3]
```diff
@@ -144,9 +144,8 @@ def check_built_config(build):
     # Build dictionary of CONFIG_NAME: y/m/n ("is not set" translates to 'n').
     configs = {}
     with open(".config", encoding='utf-8') as file:
-        for line in file:
-            line = line.strip()
-            if len(line) == 0:
+        for rawline in file:
+            if not (line := rawline.strip()):
```
Comment (nickdesaulniers) [inline check_logs.py:148]: `:=` ?

**Comment by nathanchance** [comment]
> The commit message `Fix 'ruff check --fix' for COM812` doesn't really explain what's going on. Got the warning text handy?

Yeah, sorry, that should read `Run 'ruff check --fix' for COM812. It basically just complains if each line in a multi-line call or iterable does not end in a comma

```
generate.py:38:87: COM812 [*] Trailing comma missing
```

and the `--fix` just automatically fixes it for us.

File: ruff.toml (hunk 1 above)
Comment (nathanchance) [reply ruff.toml:1]: Sure thing.

**Comment by nickdesaulniers** [comment]
> > The commit message `Fix 'ruff check --fix' for COM812` doesn't really explain what's going on. Got the warning text handy?
> 
> Yeah, sorry, that should read `Run 'ruff check --fix' for COM812`. It basically just complains if each line in a multi-line call or iterable does not end in a comma
> 
> ```
> generate.py:38:87: COM812 [*] Trailing comma missing
> ```
> 
> and the `--fix` just automatically fixes it for us.

Can we just disable that check and drop the commit that added trailing commas? They make sense IMO when defining members of an array, so that additions in later commits don't modify the `git blame`. Seeing them in parameter lists makes my eyes bleed.

**Comment by nathanchance** [comment]
Sure, I figured it was a useful check that would result in some churn up front and not too much latter but I do not care too much.

**Review by nickdesaulniers — APPROVED** [review]

**Pull request merged**